spec.md

```diff
@@ -167,6 +167,57 @@ merged are hosted in other folders.
 As some Compose file elements can both be expressed as single strings or complex objects, merges MUST apply to
 the expanded form.
 
+### Profiles
+
+Profiles allow to adjust the Compose application model for various usages and environments. A Compose
+implementation SHOULD allow the user to define a set of active profiles. The exact mechanism is implementation
+specific and MAY include command line flags, environment variables, etc.
+
+The Services top-level element supports a `profiles` attribute to define a list of named profiles. Services without
+a `profiles` attribute set MUST always be enabled. A service MUST be ignored by the compose
+implementation when none of the listed `profiles` match the active ones, unless the service is
+explicitly targeted by a command. In that case its `profiles` MUST be added to the set of active profiles.
+
+References to other services (by `links`, `extends` or shared resource syntax `service:xxx`) MUST not
+automatically enable a component that would otherwise have been ignored by active profiles.
+
+#### Illustrative example
+
+```yaml
+services:
+  foo:
+    image: foo
+  bar:
+    image: bar
+    profiles:
+      - test
+  baz:
+    image: baz
+    depends_on:
+      - bar
+    profiles:
+      - test
+  zot:
+    image: zot
+    depends_on:
+      - bar
+    profiles:
+      - debug
+```
+
+- Compose application model parsed with no profile enabled only contains the `foo` service.
+- If profile `test` is enabled, model contains the services `foo`, `bar` and `baz`.
+- If profile `debug` is enabled, model contains both `foo` and `zot` services, but not `bar` and `baz`
+  and as such the model is invalid regarding the `depends_on` constraint of `zot`.
+- If Compose implementation is executed with `bar` as explicit service to run, it will be active
```

Comment:
I think it's worth specifying that `test` profile will be enabled implicitly as well.
It illustrates the spec a bit better and means that, for example, `compose up bar` will also start `baz` (`tests` in the profile) and `foo` (always enabled).

Reply:
actually no, `compose up bar` will only start `bar` since it has no dependencies to `foo` or `baz` (both would be in the active profiles though, just not started by the command)

Reply:
This sounds inconsistent. On one hand, the PR says:

> All other top-level elements are not affected by `profiles` and are always active.

On the other hand, you suggest that:

> both would be in the active profiles though, just not started by the command

Reply:
Sorry, I can not quite follow here:
_All other top-level elements_ is intended to mean _non-service top-level elements_ like _networks_, _volumes_ etc. but `foo` and `baz` are services, so this does not apply here?

With `compose up bar` the services `foo` and `baz` will be _included_ in the compose model for the reasons you mentioned, but they will not be _started_ since only the service `bar` is targeted by the command explicitly^^

Reply:
Ah sorry, I confused myself here 🙈 Thank you for clarifying!